packages/venia-ui/lib/RootComponents/Category/category.js

```diff
@@ -26,14 +43,58 @@ const Category = props => {
 
     const [runQuery, queryResponse] = useLazyQuery(GET_CATEGORY);
     const { loading, error, data } = queryResponse;
+    const { search } = useLocation();
+    // Get "allowed" filters by intersection of schema and aggregations
+    const { data: introspectionData, error: introspectionError } = useQuery(
+        FilterIntrospectionQuery
+    );
+
+    useEffect(() => {
+        if (introspectionError) {
+            console.error(introspectionError);
+        }
+    }, [introspectionError]);
+
+    // Create a type map we can reference later to ensure we pass valid args
+    // to the graphql query.
+    // For example: { category_id: 'FilterEqualTypeInput', price: 'FilterRangeTypeInput' }
+    const filterTypeMap = useMemo(() => {
+        const typeMap = new Map();
+        if (introspectionData) {
+            introspectionData.__type.inputFields.forEach(({ name, type }) => {
+                typeMap.set(name, type.name);
+            });
+        }
+        return typeMap;
+    }, [introspectionData]);
 
     // Run the category query immediately and whenever its variable values change.
     useEffect(() => {
+        // Wait until we have the type map to fetch product data.
+        if (!filterTypeMap.size) {
+            return;
+        }
+
+        const filters = getFiltersFromSearch(search);
+
+        // Construct the filter arg object.
+        const newFilters = {};
+        filters.forEach((values, key) => {
+            newFilters[key] = getFilterInput(values, filterTypeMap.get(key));
+        });
+
+        // TODO: Category filtering on a category page is weird. How should we handle it? Currently if we don't have a filter, as in navigated from Home, we will just use the default category. If a user then filters by category, should we overwrite the category page category? It would be possible to be on the "Bottoms" category page but filter for "Tops", and see tops, but have nothing else on the page change ie breadcrumbs, etc.
+        // If a user is not filtering by category, just use the one for the page
+        // we are currently on.
+        if (!filters.get('category_id')) {
+            newFilters['category_id'] = { eq: String(id) };
+        }
```

Comment:
Spoke with @jimbo and the ideal UX does not seem possible using server side only (gql) response data.

For example, suppose you are on the "Dresses" category page and have selected to filter further by selecting "Shop the Look". One would expect that the new results are `Dresses` that are also within the `Shop the Look` category. What actually happens is that the server returns a union of all products in both categories. You end up with results like belts/accessories that happen to be in the `Shop the Look` category even though you are on the `Dresses` category page.

I thought one solution would be to take advantage of both the `eq` and `in` category filter through the following code, but it does not work:

```js
        const nextCategory = {
            eq: String(id),
            in: [String(id)]
        };
        if (newFilters['category_id']) {
            if (newFilters['category_id'].eq) {
                // If a single filter has been specified, move it to the `in` prop
                nextCategory.in.push(newFilters['category_id'].eq);
            } else if (newFilters['category_id'].in) {
                // Otherwise spread the selected categories.
                nextCategory.in.push(...newFilters['category_id'].in);
            }
        }

        // Replace the category filter with our modified filter.
        newFilters['category_id'] = nextCategory;
```

That results in a query with `category_id = { eq: "28", in: ["28", "11", "14"] }` for example, but it does not work.

I believe our only solution, without turning this ticket into a major refactor of the UX of filters, is to hide category filters entirely.